request_test.go

```diff
@@ -854,60 +984,68 @@ func TestRequest_BasicAuth(t *testing.T) {
 }
 
 func TestRequest_Host(t *testing.T) {
-	client1 := &mockClient{}
-	reporter1 := newMockReporter(t)
-
-	config1 := Config{
-		Client:   client1,
-		Reporter: reporter1,
+	cases := []struct {
+		name      string
+		method    string
+		path      string
+		url       string
+		secondURL string
```

Comment:
I suggest a couple of changes:

* remove "url" and "secondURL" fields
* add setupFunc field that will call WithHost and/or WithHeader
* add expectedHost field that will define what host to assert

I think this way it would be more clear from table what we're testing.